engines/python/setup/djl_python/properties_manager/hf_properties.py

```diff
@@ -0,0 +1,218 @@
+        elif properties['load_in_8bit']:
+            properties['quantize'] = HFQuantizeMethods.bitsandbytes8
+
+        # parsing bitsandbytes8, so it can be directly passed to lmi dist model loader.
+        if properties['quantize'] == HFQuantizeMethods.bitsandbytes8 \
+                and properties['rolling_batch'] == RollingBatchEnum.lmidist:
+            properties['quantize'] = HFQuantizeMethods.bitsandbytes
+        return properties
+
+    @root_validator()
+    def set_device(cls, properties):
+        if properties['device_id'] >= 0:
+            properties['device'] = f"cuda:{properties['device_id']}"
+        else:
+            properties['device'] = None
+        return properties
+
+    @root_validator()
+    def construct_kwargs(cls, properties):
+        kwargs = properties['kwargs']
+
+        kwargs['trust_remote_code'] = properties['trust_remote_code']
+        if properties['low_cpu_mem_usage']:
+            kwargs["low_cpu_mem_usage"] = properties['low_cpu_mem_usage']
+        if properties['revision']:
+            kwargs["revision"] = properties['revision']
+
+        if properties['rolling_batch'].value != RollingBatchEnum.disable.value:
+            if properties['output_formatter']:
+                kwargs["output_formatter"] = properties['output_formatter']
+            if properties['waiting_steps']:
+                kwargs["waiting_steps"] = properties['waiting_steps']
+
+        properties['kwargs'] = kwargs
+        return properties
+
+    @root_validator()
+    def construct_kwargs_device_map(cls, properties):
+        # https://huggingface.co/docs/accelerate/usage_guides/big_modeling#designing-a-device-map
+        kwargs = properties['kwargs']
+        if properties['device_map']:
+            kwargs["device_map"] = properties['device_map']
+            properties['device'] = None
+            logging.info(f"Using device map {properties['device_map']}")
+        elif properties[
+                'tensor_parallel_degree'] > 0 and torch.cuda.device_count(
+                ) > 0:
+            kwargs["device_map"] = "auto"
+            properties['device'] = None
+            world_size = torch.cuda.device_count()
+            assert world_size == properties['tensor_parallel_degree'], \
+                f"TP degree ({properties['tensor_parallel_degree']}) doesn't match available GPUs ({world_size})"
+            logging.info(f"Using {world_size} gpus")
+
+        properties['kwargs'] = kwargs
+        return properties
+
+    @root_validator()
+    def construct_kwargs_quantize(cls, properties):
+        kwargs = properties['kwargs']
+
+        if 'quantize' in properties and not properties['quantize']:
+            return properties
+
+        # device map is not required for lmi dist and
+        if properties['rolling_batch'] == RollingBatchEnum.lmidist or \
+                properties['rolling_batch'] == RollingBatchEnum.vllm:
+            return properties
```

Comment:
@xyang16 @KexinFeng @lanking520 
I am not validating whether device_map exists for lmi dist and vllm. And also not inserting load_in_8bit in kwargs. Kindly check if this is okay,

Reply:
device map is not important. this is only applies to scheduler and HF Accelerate path